libkbfs/data_types.go

```diff
@@ -364,8 +365,35 @@ const (
 	// folder.  Set to the empty string so that the default will be
 	// the master branch.
 	MasterBranch BranchName = ""
+
+	branchRevPrefix = "rev="
 )
 
+// MakeRevBranchName returns a branch name specifying an archive
+// branch pinned to the given revision number.
+func MakeRevBranchName(rev kbfsmd.Revision) BranchName {
+	return BranchName(fmt.Sprintf("%s%d", branchRevPrefix, rev))
+}
+
+// HasRevision returns a valid revision number and true if `bn` is a
+// revision branch.
+func (bn BranchName) HasRevision() (kbfsmd.Revision, bool) {
```

Comment:
`HasRevision` would usually imply only a bool value being returned. `MaybeGetRevision` or `RevisionIfSpecified` would be clearer.

Reply:
Or just `Revision`?

Reply:
Ok, going with `RevisionIfSpecified`, since `Revision()` doesn't convey that there's a bool involved.